pkg/arvo/sys/vane/clay.hoon

```diff
@@ -2301,6 +2323,140 @@
+    |-
+    ?~  stale
+      ..park
+    $(stale t.stale, zuz (~(del by zuz) i.stale))
+  ::
+  ::  +got-zuse: find and build zuse at kelvin
+  ::
+  ++  got-zuse
+    |=  waf=waft
+    |^  ^-  [[weft vase] _..park]
+    =/  welt=(list weft)  (sort-filter-waft waf)
+    =/  wefts=(set weft)  (silt welt)
+    ?^  cached=(try-zuse-cache welt)
+      [u.cached ..park]
+    =/  oldest=weft  (rear welt)
+    =/  =dome  dom:(~(got by dos.rom) %base)
+    =/  =aeon  let.dome
+    |-
+    ?.  (gte aeon 1)
+      (mean leaf+"clay: can't find zuse for {<(waft-to-wefts waf)>}" ~)    
+    =/  =yaki  (~(got by hut.ran) (~(got by hit.dome) aeon))
+    ?~  aeon-waft=(get-waft yaki)
+      $(aeon (dec aeon))
+    ?~  aeon-welt=(sort-filter-waft u.aeon-waft)
+      $(aeon (dec aeon))
+    =/  aeon-newest=weft  i.aeon-welt
+    :: if the newest weft for this aeon is older than
+    :: the oldest target weft, do not continue
+    ::
+    ?:  (gth num.aeon-newest num.oldest)
+      (mean leaf+"clay: gave up zuse search for {<(waft-to-wefts waf)>}" ~)    
+    ?~  (~(int in wefts) (silt aeon-welt))
+      $(aeon (dec aeon))
+    ?~  files=(get-kernel-files yaki)
+      $(aeon (dec aeon))
+    =/  zuse-core=vase  (build-zuse u.files)
+    =/  actual=weft  [%zuse !<(@ud (slap zuse-core %limb %zuse))]
+    ?.  (~(has in wefts) actual)
+      (mean leaf+"clay: zuse to /sys/kelvin mismatch at aeon {<aeon>}" ~)    
+    :-  [actual zuse-core]
+    ..park(zuz (~(put by zuz) actual zuse-core))
+    ::
+    ::  +build-zuse: compile zuse from given files
+    ::
+    ++  build-zuse
+      |=  [h=@ a=@ l=@ z=@]
+      ^-  vase
+      =/  hoon-core=vase
+        %-  (slog 'clay: compiling hoon' ~)
+        =-  (slot 7 -)
+        (road |.((slap *vase (ream h))))
+      =/  arvo-core=vase
+        %-  (slog 'clay: compiling arvo' ~)
+        =-  (slap - (ream '..part'))
+        (road |.((slap hoon-core (ream a))))
+      =/  lull-core=vase
+        %-  (slog 'clay: compiling lull' ~)
+        (road |.((slap arvo-core (rain /sys/lull/hoon l))))
+      %-  (slog 'clay: compiling zuse' ~)
+      (road |.((slap lull-core (rain /sys/zuse/hoon z))))
+    ::
+    ::  +try-zuse-cache: try get existing zuse from cache
+    ::
+    ++  try-zuse-cache
+      |=  welt=(list weft)
+      ^-  (unit [weft vase])
+      ?~  welt
+        ~
+      ?:  =(i.welt zuse+zuse)
+        (some i.welt zuse.bud)
+      ?~  zuse-core=(~(get by zuz) i.welt)
+        $(welt t.welt)
+      (some i.welt u.zuse-core)
+    ::
+    ::  +sort-filter-waft: %zuse-only and sort new-to-old
+    ::
+    ++  sort-filter-waft
+      |=  =waft
+      ^-  (list weft)
+      %+  sort
+        %+  skim  ~(tap in (waft-to-wefts waft))
+        |=(=weft =(%zuse lal.weft))
+      |=([a=weft b=weft] (lte num.a num.b))
+    ::
+    ::  +get-waft: try read /sys/kelvin from yaki
+    ::
+    ++  get-waft
+      |=  =yaki
+      ^-  (unit waft)
+      ?~  lube=(~(get by q.yaki) /sys/kelvin)
```

Comment:
it's more idiomatic to use three-letter names for local variables like these; I would suggest `lub` and `pag`